src/index.ts

```diff
@@ -73,187 +73,109 @@ export class PetraWallet implements AdapterPlugin {
       const payload = await convertV2PayloadToV1Payload(transaction.data, network)
       return "type" in payload
         ? this.signAndSubmitTransaction(payload, options)
-        : this.signAndSubmitBCSTransaction(payload, options);
```

Comment:
how come `signAndSubmitBCSTransaction` doesnt take `options`?

Reply:
`TransactionOptions` are for generating a `RawTransaction` from a `TransactionPayload`.
If you have a `RawTransaction` already, there's no need for options.

Reply:
Ah nevermind you're right, this is a bcs payload

Reply:
I think I left it that way for compatibility, but with the latest version of Petra passing `options` should work

Reply:
Ok now I'm using the new API, if options are passed